src/Components/WebView/WebView/src/PageContext.cs

```diff
@@ -17,7 +18,7 @@ namespace Microsoft.AspNetCore.Components.WebView
     /// for web views, the IPC channel is outside the page context, whereas in Blazor Server,
     /// the IPC channel is within the circuit.
     /// </summary>
-    internal class PageContext : IDisposable
+    internal class PageContext : IAsyncDisposable
     {
```

Comment:
Maybe implement both?

Reply:
We are the only consumers of this, what's the advantage of implementing both? From my point of view, the disadvantage of implementing both is that you can call `Dispose` by mistake and that will have to block without you knowing.

Reply:
Well, either block, which can be bad/dead-locky, or throw! This whole problem came about because the ServiceScope thing _throws_ if you call the wrong one.